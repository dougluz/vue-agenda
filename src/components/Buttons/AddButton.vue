<script setup lang="ts">
import IcPlus from "../icons/IcPlus.vue";

type PrimaryButtonProps = {
  small?: boolean;
};

const props = withDefaults(defineProps<PrimaryButtonProps>(), {
  small: false,
});
</script>

<template>
  <button class="add-button-base" :class="{ 'add-button-small': props.small }">
    <IcPlus /> <span>Criar contato</span>
  </button>
</template>

<style scoped>
.add-button-base {
  @apply flex items-center gap-2 justify-between bg-light-yellowish-green text-light-red font-medium text-sm rounded-[20px];
  box-shadow: inset 0 0 0 1px rgba(255, 255, 255, 0.16),
    0 0 0 0.5px rgba(0, 0, 0, 0.08), inset 0 0 0 0.5px rgba(0, 0, 0, 0.08),
    0 2px 4px 0.5px rgba(0, 0, 0, 0.16);
  padding: 0.75rem 1.375rem 0.75rem 1rem;
}

.add-button-small {
  @apply h-8;
}

.add-button-small span {
  @apply hidden md:inline-block;
}
</style>
