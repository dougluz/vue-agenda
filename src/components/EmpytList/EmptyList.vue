<script setup lang="ts">
import AddButton from "../Buttons/AddButton.vue";
import EmptyBook from "../icons/EmptyBook.vue";
</script>

<template>
  <div class="empty-list-container">
    <EmptyBook />
    <p>Nenhum contato foi criado ainda.</p>
    <AddButton />
  </div>
</template>

<style lang="css" scoped>
.empty-list-container {
  @apply flex w-full flex-1 flex-col mt-24 gap-6 items-center justify-center text-center;
}
</style>
