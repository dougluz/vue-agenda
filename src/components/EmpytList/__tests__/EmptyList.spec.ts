import AddButton from "@/components/Buttons/AddButton.vue";
import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import EmptyList from "../EmptyList.vue";

describe("Empty list tests", () => {
  it("should match the snapshot", () => {
    const wrapper = mount(EmptyList);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render the correct text", () => {
    const wrapper = mount(EmptyList);
    expect(wrapper.text()).toContain("Nenhum contato foi criado ainda.");
  });

  it("should show the button", () => {
    const wrapper = mount(EmptyList);
    const button = wrapper.findComponent(AddButton);
    expect(button.exists()).toBe(true);
  });
});
