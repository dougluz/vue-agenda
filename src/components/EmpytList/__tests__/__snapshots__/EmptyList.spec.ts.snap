// Vitest Snapshot v1

exports[`Empty list tests > should match the snapshot 1`] = `
"<div class=\\"empty-list-container\\" data-v-5ecb3237=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"237\\" height=\\"200\\" viewBox=\\"0 0 237 200\\" data-v-5ecb3237=\\"\\">
    <g fill=\\"none\\" fill-rule=\\"nonzero\\">
      <path fill=\\"#9198AF\\" d=\\"M224.741 183.673h-89.896c-6.742 0-16.345 7.143-16.345 16.327V12.245C118.5 5.51 124.016 0 130.759 0h93.982C231.484 0 237 5.51 237 12.245v159.184c0 6.734-5.516 12.244-12.259 12.244z\\"></path>
      <path fill=\\"#C0C3D2\\" d=\\"M12.259 183.673h89.896c6.742 0 16.345 7.143 16.345 16.327V12.245C118.5 5.51 112.984 0 106.241 0H12.26C5.516 0 0 5.51 0 12.245v159.184c0 6.734 5.516 12.244 12.259 12.244z\\"></path>
    </g>
  </svg>
  <p data-v-5ecb3237=\\"\\">Nenhum contato foi criado ainda.</p><button class=\\"add-button\\" data-v-187b35fa=\\"\\" data-v-5ecb3237=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" data-v-187b35fa=\\"\\">
      <path fill=\\"#FA7268\\" fill-rule=\\"nonzero\\" d=\\"M16 7.238H8.762V0H7.238v7.238H0v1.524h7.238V16h1.524V8.762H16z\\"></path>
    </svg> Criar contato </button>
</div>"
`;
